### PR TITLE
Tn10 delete chats id emails

### PR DESCRIPTION
**Pull request by bvbps**

### CODE REVIEW

File: app/repositories/slick/implementations/SlickChatsRepository.scala [hunk 1]
```diff
@@ -320,6 +320,66 @@ class SlickChatsRepository @Inject() (db: Database)(implicit executionContext: E
...
+      optionUserChat <- getDraftsUserChat(userId, chatId).result.headOption
+      optionDraft <- getDraftEmailQuery(chatId, emailId).result.headOption
+
+      fromAddressIdQuery = getEmailAddressesQuery(chatId, emailId).filter(_.participantType === "from").map(_.addressId)
+
+      optionFromUserId <- UsersTable.all
+        .filter(userRow => userRow.addressId.in(fromAddressIdQuery) && userRow.userId === userId)
+        .result.headOption
+
+      condition = List(optionUserChat, optionDraft, optionFromUserId).forall(_.isDefined)
```
Comment (IvoSaraiva) [inline app/repositories/slick/implementations/SlickChatsRepository.scala]: Just remove the condition variable and put this on the yield.

File: app/repositories/slick/implementations/SlickChatsRepository.scala [hunk 2]
```diff
@@ -387,7 +447,8 @@ class SlickChatsRepository @Inject() (db: Database)(implicit executionContext: E
     if (addresses.nonEmpty) {
       for {
         updateReceiversChats <- updateReceiversUserChatsToInbox(chatId, addresses)
-        updateEmailStatus <- EmailsTable.all.filter(_.emailId === emailId).map(_.sent).update(1)
+        updateEmailStatus <- EmailsTable.all
+          .filter(_.emailId === emailId).map(emailRow => (emailRow.date, emailRow.sent)).update(DateUtils.getCurrentDate, 1)
         updateSenderChat <- UserChatsTable.userEmailWasSent(senderUserId, chatId)
       } yield updateReceiversChats.sum + updateEmailStatus + updateSenderChat
     } else DBIO.successful(0)
```
Comment (IvoSaraiva) [inline app/repositories/slick/implementations/SlickChatsRepository.scala]: Should this be "transactionally"?

File: app/repositories/slick/implementations/SlickChatsRepository.scala [hunk 3]
```diff
@@ -670,7 +731,7 @@ class SlickChatsRepository @Inject() (db: Database)(implicit executionContext: E
     (for {
       subject <- ChatsTable.all.filter(_.chatId === chatId).map(_.subject)
       _ <- UserChatsTable.all.filter(userChatRow => userChatRow.chatId === chatId && userChatRow.userId === userId &&
-        (userChatRow.inbox === 1 || userChatRow.sent === 1 || userChatRow.draft === 1 || userChatRow.trash === 1))
+        (userChatRow.inbox === 1 || userChatRow.sent === 1 || userChatRow.draft >= 1 || userChatRow.trash === 1))
```
Comment (IvoSaraiva) [inline app/repositories/slick/implementations/SlickChatsRepository.scala]: Can draft be bigger than 1?

File: test/repositories/slick/implementations/ChatsRepositorySpec.scala [hunk 4]
```diff
@@ -866,4 +866,77 @@ class ChatsRepositorySpec extends AsyncWordSpec with MustMatchers with OptionVal
...
+    val validDraftEmailId = "fe4ff891-144a-4f61-af35-6d4a5ec76314"
+
+    "not delete the draft if the user is not the owner/sender of the email" in {
+      val notAllowedUserId = "adcd6348-658a-4866-93c5-7e6d32271d8d"
+      for {
+        deleteDraft <- chatsRep.deleteDraft(validChatId, validDraftEmailId, notAllowedUserId)
+
+        emailRow <- db.run(EmailsTable.all.filter(_.emailId === validDraftEmailId).result.headOption)
+        emailAddressesRows <- db.run(EmailAddressesTable.all.filter(_.emailId === validDraftEmailId).result.headOption)
+        //attachmentsRows <- db.run(AttachmentsTable.all.filter(_.emailId === validSentEmailId).result.headOption)    //this example does not have attachments
```
Comment (IvoSaraiva) [inline test/repositories/slick/implementations/ChatsRepositorySpec.scala]: Commented code?

File: app/repositories/slick/implementations/SlickChatsRepository.scala (hunk 3 above)
Comment (bvbps) [reply app/repositories/slick/implementations/SlickChatsRepository.scala]: Yes, we are allowing users to have more than one draft per chat

File: app/repositories/slick/implementations/SlickChatsRepository.scala (hunk 2 above)
Comment (bvbps) [reply app/repositories/slick/implementations/SlickChatsRepository.scala]: Yes, it should. Thanks

**Pull request merged**